src/org/omegat/core/search/Searcher.java

```diff
@@ -183,6 +183,11 @@ public void search() throws TranslationException, PatternSyntaxException, IOExce
                 text = text.replaceAll(" ", "( |\u00A0)");
             }
 
+            // handle ZenHan insensitive match
+            if (expression.zenHanInsensitive) {
+                text = StringUtil.replaceZenHan(text);
```

Comment:
You should normalize the width of the search string (just once, of course) in addition to the `text.`